### PR TITLE
Proof PR: Refactor Makefile

### PR DESCRIPTION
**Pull request by greg0ire**
Make it usable for everyone, not just the CI.
Don't merge, but do review

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,22 +2,41 @@
...
+.PHONY: lint-yaml
+
+lint-xml:
 	find . \( -name '*.xml' -or -name '*.xliff' \) \
 		-not -path './vendor/*' -not -path './Resources/public/vendor/*' \
 		| xargs -I'{}' xmllint --encode UTF-8 --output '{}' --format '{}'
+.PHONY: lint-xml
+
+lint-php:
+	TARGET_PHP_VERSION=$(shell grep '"php":' composer.json|cut -d '"' -f4|cut -b 2-5) \
```
Comment (greg0ire) [inline Makefile]: That's the dirty part, open to other solutions

File: Makefile [hunk 2]
```diff
@@ -2,22 +2,41 @@
 #
 # It's auto-generated by sonata-project/dev-kit package.
 
-.PHONY: test docs
-
 all:
 	@echo "Please choose a task."
+.PHONY: all
+
+lint: lint-composer lint-yaml lint-composer lint-xml lint-php
```
Comment (greg0ire) [inline Makefile]: This means people can run each linter separately.

File: Makefile [hunk 3]
```diff
@@ -2,22 +2,41 @@
...
+
+lint-yaml:
 	find . -name '*.yml' -not -path './vendor/*' -not -path './Resources/public/vendor/*' | xargs yaml-lint
+
+.PHONY: lint-yaml
+
+lint-xml:
 	find . \( -name '*.xml' -or -name '*.xliff' \) \
 		-not -path './vendor/*' -not -path './Resources/public/vendor/*' \
 		| xargs -I'{}' xmllint --encode UTF-8 --output '{}' --format '{}'
```
Comment (greg0ire) [inline Makefile]: I think I have to change that one, it's a fixer, not a linter I think

**Comment by greg0ire** [comment]
Proof build with invalid php: https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/312613957

File: Makefile (hunk 2 above)
Comment (greg0ire) [reply Makefile]: make -j <put your number of processors here> lint FTW

File: Makefile [hunk 4]
```diff
@@ -2,22 +2,58 @@
...
-		| xargs -I'{}' xmllint --encode UTF-8 --output '{}' --format '{}'
+
+.PHONY: lint-yaml
+
+lint-xml:
+	find src \( -name '*.xml' -or -name '*.xliff' \) \
+		-not -path './src/Resources/public/vendor/*' \
+		| while read xmlFile; \
+	do \
+		XMLLINT_INDENT='    ' xmllint --encode UTF-8 --format "$$xmlFile"|diff - "$$xmlFile"; \
```
Comment (greg0ire) [inline Makefile]: Too bad Travis does not have the `--color` diff option... maybe on the 7.1 image?

File: .php_cs.dist [hunk 5]
```diff
@@ -39,21 +46,21 @@ $rules = [
 //    'php_unit_strict' => true,
 ];
 
-if (\PHP_VERSION_ID >= 50600) {
+if (version_compare($targetPhpVersion, '5.6', '>=')) {
```
Comment (jordisala1991) [inline .php_cs.dist]: Imo it makes no sense to have conditionals on this file. A branch should have its own ruleset and it should be regardless of what version of php do you use to lint it

Comment (greg0ire) [reply .php_cs.dist]: So you would use twig to generate that file, right? I think this is a better solution indeed, what I did is dirty.

Comment (Soullivaneuh) [reply .php_cs.dist]: @jordisala1991 I don't see the issue to have conditional statement. The configuration is the PHP file, we can take benefit from it.

BTW, moving it to a twig file on dev-kit is just make it dynamic else where.

Comment (greg0ire) [reply .php_cs.dist]: Yeah but see how I'm getting `$targetPhpVersion` : yuck! plus now it requires people only ever use `make` to run the cs-fixer

**Comment by jordisala1991** [comment]
In theory this can be closed, already merged on dev-kit. Only missing deploy

**Pull request closed without merging**